### ISSUE TITLE
investigate using puppet 'environments'

### ISSUE THREAD

**Issue by binary1230** [magfest/ubersystem-deploy#22]
We don't currently have a good way, even with hiera, to combine configs in this way:

final production config = common config + production-specific config
or
staging config = common config + testing-specific config

puppet has a config option called 'environments' that may be what we're looking for.

also, using an External Node Classifier (ENC) might be able to do stuff with this kind of granularity.
basically our ENC can say 'this node is a production (vs testing) node, so set the environment=production'

Then you could include yaml hiera files like
- common.yaml
- common-{%environment}.yaml

so we could have a common-production.yaml and a common-staging.yaml

something like that.

**Comment by binary1230** [comment]
did this
